### PR TITLE
Fix a bug where Terminal may crash or hang at shutdown

### LINKED ISSUE #1380: [~75% IMPACT] Terminal still crashes on exit because of XamlApplication
The windows will stay like this for a few seconds.It should close immediately.
屏幕截图(59)
What'more , I have found a closed issue which says that this bug should have been fixed.

I'm now using 10.0.18362.175 ,and my copy of Terminal is from the Store.

### PR DESCRIPTION
**Pull request by ocalvo**
## Summary of the Pull Request
Fix issue [1380](https://github.com/microsoft/terminal/issues/1380)
 
## References
[1380](https://github.com/microsoft/terminal/issues/1380)

## PR Checklist
* [X] Closes #1380
* [X] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Detailed Description of the Pull Request / Additional comments
The order of destruction is incorrect:
Construction: 1. XamlApp, 2. XamlIsland
Destruction: 1. XamlApp, 2. XamlIsland, it should be 1. Xamlsland, 2. XamlApp

## Validation Steps Performed
Manually debugged that the Xaml core is de-initialized correctly.

### CODE REVIEW

File: src/cascadia/WindowsTerminal/AppHost.cpp [hunk 1]
```diff
@@ -41,7 +41,9 @@ AppHost::AppHost() noexcept :
 
 AppHost::~AppHost()
 {
+    _window = nullptr;
```
Comment (DHowett-MSFT) [inline src/cascadia/WindowsTerminal/AppHost.cpp]:
```suggestion
    // destruction order is important for proper teardown here
    _window = nullptr;
```

**Comment by DHowett-MSFT** [comment]
@msftbot merge this in 1 hour

File: src/cascadia/WindowsTerminal/WindowsTerminal.vcxproj [hunk 2]
```diff
@@ -90,6 +90,16 @@
       <DeploymentContent>true</DeploymentContent>
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </NativeReferenceFile>
+
```
Comment (DHowett-MSFT) [inline src/cascadia/WindowsTerminal/WindowsTerminal.vcxproj]: What happened here? Does this need to be added again? Is github acting weird?

Comment (ocalvo) [reply src/cascadia/WindowsTerminal/WindowsTerminal.vcxproj]: I don't know what happened here, I did a git rebase to make sure.
I am investigating

**Pull request merged**